### ISSUE TITLE
How to recall initState method in Indexed Stack children widget

### ISSUE THREAD

**Issue by myoganugraha** [flutter/flutter#48110]
Hello, i create blank screen with bottom navigation bar and indexed stack as a body of the scaffold.
And for every indexed stack's children, i've implement flutter_bloc and called it on initState() function.

But i finally found issue for me, lol. The initState() function of every children called only on first time, 
if i open it from bottom navbar, the initState() function didn't call again. I think it was because i'm using indexed stack.

so, how to call initState() even using indexed stack

thanks

**Comment by julianbraasch** [comment]
```initState``` is meant to be called only once. Try placing your code in ```didChangeDependencies``` and see if that helps.

**Comment by myoganugraha** [comment]
@julianbraasch yup, usually the page will finish when change to other, but when i'm using indexed stack looks like different haha

okay, thank you, i'll try it soon

**Comment by julianbraasch** [comment]
Pass a ```UniqueKey``` as ```key``` to your widget, this should force the lifecycle to start over

**Comment by iapicca** [comment]
Hi @myoganugraha 
does @julianbraasch 's proposed solution fix the issue?
thank you

**Comment by myoganugraha** [comment]
> Pass a `UniqueKey` as `key` to your widget, this should force the lifecycle to start over

well, i'll try it soon



> Hi @myoganugraha
> does @julianbraasch 's proposed solution fix the issue?
> thank you

i can't confirm it now, i'm try implement the solution asap :D

**Comment by myoganugraha** [comment]
> Pass a `UniqueKey` as `key` to your widget, this should force the lifecycle to start over

this should be confirmed answer for my problem
thanks @julianbraasch

**Comment by rodriigovieira** [comment]
@julianbraasch I added a UniqueKey to all widgets that are in the IndexedStack, and also to the IndexedStack itself, but it did not work. Could you help me?

@myoganugraha what changes did you do in your code to make it work?

**Comment by csongorkeller** [comment]
Hi, @myoganugraha I'm having the same issue. Basically, I have 2 pages in indexed stack and what I want is somehow force flutter to rebuild page 2 and make the API calls there. I've tried to pass a Unique key as scaffold key but didn't help. Can you share your solution, or give some more hint on how it could be implemented?

How do you pass a Unique key to your widget to trigger being rebuilt? 

Many thanks in advance

**Comment by iapicca** [comment]
@csongorkeller 
can you please open a new issue providing your `flutter doctor -v` and a [minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example)

**Comment by Charlesvandamme** [comment]
> que key to your widget to trigger being rebuilt?
> 
> Many thanks in advance



> Hi, @myoganugraha I'm having the same issue. Basically, I have 2 pages in indexed stack and what I want is somehow force flutter to rebuild page 2 and make the API calls there. I've tried to pass a Unique key as scaffold key but didn't help. Can you share your solution, or give some more hint on how it could be implemented?
> 
> How do you pass a Unique key to your widget to trigger being rebuilt?
> 
> Many thanks in advance

You don't pass the unique key in the initstate, but rather to the newly created Widget when the button is pressed. In my case, the page I want to rebuild has index 0 (Homepage), so I do the following in my 'onTap' function: 

`  
    if (index == 0) {

      setState(() {

        screens.removeAt(0);

        screens.insert(0, new HomePage(key: UniqueKey()));

        tabIndex = index;
      });
    }

**Comment by albertms10** [comment]
> Pass a `UniqueKey` as `key` to your widget, this should force the lifecycle to start over

You saved my night. Thank you so much! 😄
